### ISSUE TITLE
Release vs Test container registry logic should be revisited

### ISSUE THREAD

**Issue by rickducott** [solo-io/build#13]
* Don't think we should automatically push to release if test isn't defined
* On release builds, we still may want to push test images prior to running regression tests. If those fail, we don't want to have published actual release images. So we should always use the test registry, and only ship to the release registry on release builds that have it set.

**Comment by mitchdraft** [comment]
Good idea, we could also do the same for helm resources and anything else that produces release artifacts as long as we can prove that the artifacts that we copy from the "test" location to the "release" location are identical.

Then maybe our "release" process is just a matter of copying resources that have been labeled as a release from the test locations to their release locations.

This could also include binaries, then we could provide them through solo.io, rather than github
